pg_sparse/src/api/mod.rs

```diff
@@ -0,0 +1,118 @@
+use pgrx::*;
+use std::collections::HashMap;
+
+use crate::sparse_index::sparse::Sparse;
+
+#[pg_extern(immutable, strict, parallel_safe)]
+pub fn sparse_cosine_distance(left: Sparse, right: Sparse) -> f32 {
```

Comment:
What is the purpose of this? The cosine distance is calculated in hnswlib, so it's redundant to have it here. Is it exposed as a user function?

Reply:
This is for the purpose of a sequential scan. Hnswlib is only invoked when postgres decides that an index scan is optimal.